torch/lib/THD/base/ChannelUtils.hpp

```diff
@@ -10,6 +10,48 @@
 #include <tuple>
 #include <vector>
 
+
+inline void hash_combine(std::size_t& seed) { }
+
+template <typename T, typename... Rest>
+inline void hash_combine(std::size_t& seed, const T& v, Rest... rest) {
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+  hash_combine(seed, rest...);
+}
+
+#define SINGLE_ARG(...) __VA_ARGS__
```

Comment:
I think it might be simpler to use `using some_type = std::tuple<...>` and pass that alias